source/common/stats/utility.h

```diff
@@ -38,7 +56,7 @@ class DynamicName : public absl::string_view {
  * hide the memory management of the joined storage, and they allow easier
  * co-mingling of symbolic and dynamic stat-name components.
  */
-using Element = absl::variant<StatName, DynamicName>;
+using Element = absl::variant<StatName, DynamicName, DynamicSavedName>;
```

Comment:
This is not allowed:

https://en.cppreference.com/w/cpp/utility/variant
> As with unions, if a variant holds a value of some object type T, the object representation of T is allocated directly within the object representation of the variant itself. Variant is not allowed to allocate additional (dynamic) memory.

while `std::string` allocates additional memory to hold the string thus it may be leaked.

Reply:
I interpret it as "like unions a variant allocates enough memory to contain a value of the biggest type T thus doesn't need to allocate more (dynamically) when a value of a smaller type gets replaced with another value of a bigger type".
As with sts::string a variant needs to store its control structure only which doesn't change. And unlike unions variants call the destructors of Ts. So, no leaks are expected.

Reply:
Ah that makes more sense, my bad.